shiny/_main.py

```diff
@@ -220,12 +232,17 @@ def run_app(
 
     log_config: Dict[str, Any] = copy.deepcopy(uvicorn.config.LOGGING_CONFIG)
 
-    if reload and app_dir is not None:
-        reload_dirs = [app_dir]
-    else:
+    if reload_dirs is None:
         reload_dirs = []
-
     if reload:
+        # Always watch the app_dir
+        if app_dir:
+            reload_dirs.append(app_dir)
+        # For developers of Shiny itself; autoreload the app when Shiny package changes
+        if os.getenv("SHINY_PKG_AUTORELOAD"):
+            shinypath = Path(inspect.getfile(shiny)).parent
+            reload_dirs.append(str(shinypath))
```

Comment:
Super neat! So are you `export`ing this in your `.profile`, or something better?

Reply:
Just in .profile.